components/suggestion/at_mention_provider/at_mention_provider.tsx

```diff
@@ -7,39 +7,78 @@ import {getSuggestionsSplitBy, getSuggestionsSplitByMultiple} from 'mattermost-r
 
 import {Constants} from 'utils/constants';
 
-import Provider from '../provider.jsx';
+import Provider, {ResultCallbackParams} from '../provider';
 
-import AtMentionSuggestion from './at_mention_suggestion.jsx';
+import {UserProfile, UserProfileWithLastViewAt} from 'mattermost-redux/types/users';
+
+import {Group} from 'mattermost-redux/types/groups';
+
+import AtMentionSuggestion from './at_mention_suggestion';
+
+interface Props {
+    currentUserId: string;
+    profilesInChannel: UserProfileWithLastViewAt[];
+    autocompleteUsersInChannel: (prefix: string) => Promise<UserProfile[]>;
+    useChannelMentions: boolean;
+    autocompleteGroups: Group[];
+    searchAssociatedGroupsForReference: (prefix: string) => Promise<{data: Group[]}>;
+    priorityProfiles: UserProfileWithLastViewAt[];
+}
+
+export interface LocalMember extends UserProfileWithLastViewAt {
+    type: string;
+    isCurrentUser?: boolean;
+}
+
+export interface LocalGroup extends Group {
+    type: string;
+}
+
+export interface SpecialMention {
+    username: string;
+    type: string;
+}
+
+export type Item = LocalMember & LocalGroup & SpecialMention;
 
 // The AtMentionProvider provides matches for at mentions, including @here, @channel, @all,
 // users in the channel and users not in the channel. It mixes together results from the local
 // store with results fetch from the server.
 export default class AtMentionProvider extends Provider {
-    constructor(props) {
+    data: any;
+    lastCompletedWord: string;
+    lastPrefixWithNoResults: string;
+    triggerCharacter: string;
+
+    currentUserId: string;
+    profilesInChannel: UserProfileWithLastViewAt[];
+    autocompleteUsersInChannel: (prefix: string) => Promise<UserProfile[]>;
+    useChannelMentions: boolean;
+    autocompleteGroups: Group[];
+    searchAssociatedGroupsForReference: (prefix: string) => Promise<{data: Group[]}>;
+    priorityProfiles: UserProfile[];
+
+    constructor(props: Props) {
         super();
 
-        this.setProps(props);
-
         this.data = null;
         this.lastCompletedWord = '';
         this.lastPrefixWithNoResults = '';
         this.triggerCharacter = '@';
-    }
 
-    // setProps gives the provider additional context for matching pretexts. Ideally this would
-    // just be something akin to a connected component with access to the store itself.
-    setProps({currentUserId, profilesInChannel, autocompleteUsersInChannel, useChannelMentions, autocompleteGroups, searchAssociatedGroupsForReference, priorityProfiles}) {
```

Comment:
We need to keep this `setProps` method here.

```ts
setProps(props: Props) {
```

Reply:
Fixed